### PR TITLE
feat: add ml.llm.Claude3TextGenerator model

### PR DESCRIPTION
**Pull request by GarrettWu**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery-dataframes/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes b/359901494 🦕

### CODE REVIEW

**Comment by GarrettWu** [comment]
e2e is failing since bigframes-load-testing hasn't been allowlisted.

File: bigframes/ml/llm.py [hunk 1]
```diff
@@ -54,6 +54,13 @@
     _GEMINI_1P5_PRO_FLASH_PREVIEW_ENDPOINT,
 )
 
+_CLAUDE_3_SONNET_ENDPOINT = "claude-3-sonnet"
```
Comment (jiaxunwu) [inline bigframes/ml/llm.py:57]: Could you add all BQML-supported models?

File: bigframes/ml/llm.py [hunk 2]
```diff
@@ -860,3 +867,213 @@ def to_gbq(self, model_name: str, replace: bool = False) -> GeminiTextGenerator:
...
+
+    .. note::
+
+        This product or feature is subject to the "Pre-GA Offerings Terms" in the General Service Terms section of the
+        Service Specific Terms(https://cloud.google.com/terms/service-terms#1). Pre-GA products and features are available "as is"
+        and might have limited support. For more information, see the launch stage descriptions
+        (https://cloud.google.com/products#product-launch-stages).
+
+    Args:
+        model_name (str, Default to "claude-3-sonnet"):
```
Comment (shobsi) [inline bigframes/ml/llm.py:884]: [nit] "Defaults to ..."?

File: bigframes/ml/llm.py [hunk 3]
```diff
@@ -860,3 +867,213 @@ def to_gbq(self, model_name: str, replace: bool = False) -> GeminiTextGenerator:
 
         new_model = self._bqml_model.copy(model_name, replace)
         return new_model.session.read_gbq_model(model_name)
+
+
+@log_adapter.class_logger
+class Claude3TextGenerator(base.BaseEstimator):
+    """Claude3 text generator LLM model.
```
Comment (shobsi) [inline bigframes/ml/llm.py:874]: Looks like "Consumer Procurement Entitlement Manager Identity and Access Management (IAM) role" is an additional requirement https://cloud.google.com/vertex-ai/generative-ai/docs/partner-models/use-partner-models#set-permissions, we should document this in the class docstring and after the release in the reference docs https://cloud.google.com/bigquery/docs/use-bigquery-dataframes#remote-models

File: bigframes/ml/llm.py (hunk 2 above)
Comment (GarrettWu) [reply bigframes/ml/llm.py:1056]: We use "Default to" in all the APIs... Keeping it for now.

File: bigframes/ml/llm.py (hunk 3 above)
Comment (GarrettWu) [reply bigframes/ml/llm.py:1038]: Updated.

**Review by jiaxunwu — APPROVED** [review]

File: tests/system/conftest.py [hunk 4]
```diff
@@ -145,6 +145,16 @@ def session() -> Generator[bigframes.Session, None, None]:
     session.close()  # close generated session at cleanup time
 
 
+@pytest.fixture(scope="session")
+def session_us_east5() -> Generator[bigframes.Session, None, None]:
```
Comment (shobsi) [inline tests/system/conftest.py:149]: Did we use it anywhere?

File: bigframes/ml/llm.py (hunk 1 above)
Comment (shobsi) [reply bigframes/ml/llm.py:64]: Sorry, I thought you were going to add in this PR itself. Are you going to send another PR for this?

Comment (GarrettWu) [reply bigframes/ml/llm.py:64]: Yes. need to setup the connection for other regions.

File: tests/system/conftest.py (hunk 4 above)
Comment (GarrettWu) [reply tests/system/conftest.py:149]: forgot to remove in this PR, but will use in PR of adding tests.

**Pull request merged**